workflow/any.go

```diff
@@ -46,8 +46,8 @@ func (a *anySchemaWithExpressions) Serialize(data any) (any, error) {
 	return a.checkAndConvert(data)
 }
 
-func (a *anySchemaWithExpressions) ApplyScope(scope schema.Scope, namespace string) {
-	a.anySchema.ApplyScope(scope, namespace)
+func (a *anySchemaWithExpressions) ApplyNamespace(objects map[string]*schema.ObjectSchema, namespace string) {
+	a.anySchema.ApplyNamespace(objects, namespace)
 }
```

Comment:
Yes!...at least to the extent that Go and the current SDK seem to support.  😐

However, ideally, we would DRY out `checkAndConvert()` too (so that it ends up looking just like `ValidateCompatibility()`):
```go
func (a *anySchemaWithExpressions) checkAndConvert(data any) (any, error) {
	if _, ok := data.(expressions.Expression); ok {
		return data, nil
	}
	return a.AnySchema.checkAndConvert(data)
}
```
but I don't know whether Go's scoping rules will permit that.  (And, of course, this assumes that there is no way to have a slice or map of `Expression`'s...can those exist??)

Reply:
The other possibility is to hoist the special case for `Expression` out of `checkAndConvert()` into its callers and modify them to call their corresponding receiver on the embedded type for things which aren't `Expression`'s.  That would be worth it (it's a small change, and it makes all the receivers consistent with `ValidateCompatibility()`) if it allows us to get rid of the `checkAndConvert()` receiver on `anySchemaWithExpressions`.

(I'd really like to see the duplication in `checkAndConvert()` removed, if feasible, because it's a big piece of code, and we have to remember that it exists in two places, which is always fragile.)

Reply:
I can't change the logic, because if it calls the parent function, it will use that class's logic for the items of the slices and maps, which breaks it.
But I was able to make it briefer by letting AnySchema handle the error case, which meant I could use the default case for everything else.

Anything else, if it is even possible, is stretching the scope of this PR.

Reply:
Your update is excellent.  The problem here was that I missed the fact that a workflow can construct lists or objects whose values are expressions, and therefore `anySchemaWithExpressions` needs to be able to catch those as they go by, and to do that we need the overrider for `checkAndConvert()` to be called which, of course, is prohibitively difficult to effect in Go.

Thanks for indulging me, I think the result is a huge improvement.